### PR TITLE
feat: Fix lints reported by ajv

### PR DESCRIPTION
**Pull request by l0b0**

### CODE REVIEW

**Comment by m-mohr** [comment]
IMHO these seem superfluous because the type is already defined in the parent schema. Why is ajv complaining about them?

**Comment by l0b0** [comment]
From [the documentation](https://ajv.js.org/strict-mode.html#require-applicable-types):

> the problem here is that any value that is not an object would be valid against [a schema which doesn't explicitly state the type] - this is rarely intentional.

As it turns out, there are also a whole bunch of even core schema properties which don't specify the `type` explicitly, so I'd be a bit worried that they might actually be unspecified all the way up.

In any case, working out whether it's safe to omit the `type` in a specific place would be quite tricky, so I agree with ajv that it's better to be explicit.

**Comment by m-mohr** [comment]
Interesting. But if you validate against two schemas anyway (one with type, one without), one of them will complain about the wrong type, so you have validated correctly. I don't think this is really an issue tbh. We need to verify this though.

**Comment by l0b0** [comment]
> Interesting. But if you validate against two schemas anyway (one with type, one without), one of them will complain about the wrong type, so you have validated correctly. I don't think this is really an issue tbh. We need to verify this though.

Considering how hard it would be to verify thoroughly I'd probably go for the better-safe-than-sorry approach, even if it's a little more verbose.

File: json-schema/schema.json
Comment (m-mohr) [inline json-schema/schema.json:201]: This can easily be enforced in the object that also contains the anyOf  so that we at least don't need to repeat these 4 times.

**Review by m-mohr — APPROVED** [review]
I'll approve and leave it up to the extension maintainer to decide and merge. It doesn't break anything, but it also doesn't fix anything. These object types are superfluous as they are all enforced in the core schemas anyway, I've checked them one by one. Left one comment to reduce the amount of type annotations.

**Comment by emmanuelmathot** [comment]
I approve and merge because it does not break anything but I am sharing the same view as @m-mohr. Those repeated annotation do not really enforce anything more and just add more maintenance for future changes.

**Pull request merged**
